src/Auth0.OidcClient.Core/Tokens/IdTokenRequirements.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+
+namespace Auth0.OidcClient.Tokens
+{
+    /// <summary>
+    /// Identity token validation requirements for use with <see cref="IdTokenValidator"/>.
+    /// </summary>
+    internal class IdTokenRequirements
+    {
+        /// <summary>
+        /// Required issuer (iss) the token must be from.
+        /// </summary>
+        public string Issuer;
+
+        /// <summary>
+        /// Required audience (aud) the token must be for.
+        /// </summary>
+        public string Audience;
+
+        /// <summary>
+        /// Optional one-time nonce the token must be issued in response to.
+        /// </summary>
+        public string Nonce;
+
+        /// <summary>
+        /// Optional maximum age of the user authentication underpinning the token.
+        /// </summary>
+        public TimeSpan? MaxAge = null;
+
+        /// <summary>
+        /// Whether an identity token is required. Defaults to true.
+        /// </summary>
+        public bool TokenIsRequired = true;
+
+        /// <summary>
+        /// Amount of leeway to allow in validating date and time claims in order to allow some clock variance
```

Comment:
```suggestion
        /// Amount of leeway in seconds to allow in validating date and time claims in order to allow some clock variance
```

Reply:
It's not actually seconds, it's a .NET TimeSpan class which allows you to create it in minutes, seconds, microseconds, hours...

Reply:
e.g. `Leeway = TimeSpan.FromSeconds(60)` or `Leeway = TimeSpan.FromMinutes(1)`